internal/api/keppel/accounts.go

```diff
@@ -542,6 +544,66 @@ func (a *API) handlePutAccount(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		// Copy PlatformFilter when creating an account with the Replication Policy on_first_use
+		if req.Account.ReplicationPolicy != nil {
+			rp := *req.Account.ReplicationPolicy
+			if rp.Strategy == "on_first_use" {
+				var peer keppel.Peer
+				err := a.db.SelectOne(&peer, `SELECT * FROM peers WHERE hostname = $1`, rp.UpstreamPeerHostName)
+				if err == sql.ErrNoRows {
+					http.Error(w, fmt.Sprintf(`unknown peer registry: %q`, rp.UpstreamPeerHostName), http.StatusUnprocessableEntity)
+					return
+				}
+				if respondwith.ErrorText(w, err) {
+					return
+				}
+
+				reqURL := fmt.Sprintf("https://%s/peer/v1/account-filter/%s", accountToCreate.UpstreamPeerHostName, accountToCreate.Name)
+				authReq, err := http.NewRequest("GET", reqURL, nil)
+				if respondwith.ErrorText(w, err) {
+					return
+				}
+				//TODO: a.cfg.OurReplicationUsername = fmt.Sprintf("replication@%s", a.cfg.APIPublicURL.Hostname())
+				authReq.Header.Set("Authorization", keppel.BuildBasicAuthHeader(fmt.Sprintf("replication@%s", a.cfg.APIPublicURL.Hostname()), peer.OurPassword))
+
+				resp, err := http.DefaultClient.Do(authReq)
+				if err != nil {
+					http.Error(w, "could not validate credentials: "+err.Error(), http.StatusUnauthorized)
+					return
+				}
+				if resp.StatusCode != http.StatusOK {
+					http.Error(w, "could not validate credentials: expected 200 OK, but got "+resp.Status, http.StatusUnauthorized)
+					return
+				}
+
+				respBodyBytes, err := ioutil.ReadAll(resp.Body)
+				if err == nil {
+					err = resp.Body.Close()
+				} else {
+					resp.Body.Close()
+				}
+				if err != nil {
+					return
+				}
```

Comment:
Each branch that does `return` should have rendered a HTTP response beforehand:
```suggestion
				if respondwith.ErrorText(w, err) {
					return
				}
```